### ISSUE TITLE
Error: Key "plugins": This appears to be in eslintrc format (array of strings) rather than flat config format (object).

### ISSUE THREAD

**Issue by brianjenkins94** [eslint-stylistic/eslint-stylistic#230]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

This isn't strictly speaking an issue with `eslint-stylistic` and may also require a bug report to `dbaeumer.vscode-eslint` but I think it would be worthwhile to have this documented since it may influence the content in the [Migration Guide](https://eslint.style/guide/migration).

This config:
```ts
// eslint.config.js
import eslint from "@eslint/js";
import typescriptParser from "@typescript-eslint/parser";
import typescriptPlugin from "@typescript-eslint/eslint-plugin";

export default [
	{
		"plugins": [
			"@typescript-eslint"
		],
```

Throws this error in the Output > ESLint tab in Visual Studio Code:

```
Error: Key "plugins": This appears to be in eslintrc format (array of strings) rather than flat config format (object).
```

In remediating this, I had to play a bit of guess and check with how the individual package guides are currently written, since [the guides](https://eslint.style/packages/default) suggest:

```js
// .eslintrc.js
module.exports = {
  plugins: [
    '@stylistic'
  ]
```

But what I ultimately needed was:

```ts
import eslint from "@eslint/js";
import typescriptParser from "@typescript-eslint/parser";
import typescriptPlugin from "@typescript-eslint/eslint-plugin";
import stylisticPlugin from "@stylistic/eslint-plugin";

export default [
	{
		"plugins": {
			"@typescript-eslint": typescriptPlugin,
			"@stylistic": stylisticPlugin
		},
```

But since I'm only planning on linting `*.ts` files, I'm not sure if it would be better to import `@stylistic/eslint-plugin-ts` instead.

Generally speaking, I'm not sure when it makes sense to specify rules with `@stylistic/ts/` vs `@stylistic/`.

### Reproduction

N/A

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by antfu** [comment]
In legacy config format, `plugins` takes an array of string; In Flat config it takes an object: https://eslint.org/docs/latest/use/configure/configuration-files-new#configuration-objects

The document is not wrong, as we indicated it's `// .eslintrc.js`. While indeed we could improve the docs to also mentions about Flat config usage

**Comment by brianjenkins94** [comment]
Oh, I incorrectly assumed that the package docs were using the flat config format.
